handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -434,10 +436,13 @@ private boolean hasOutputChanged(ChannelHandlerContext ctx, boolean first) {
             if (buf != null) {
                 int messageHashCode = System.identityHashCode(buf.current());
                 long pendingWriteBytes = buf.totalPendingWriteBytes();
+                long flushProgress = buf.currentProgress();
 
-                if (messageHashCode != lastMessageHashCode || pendingWriteBytes != lastPendingWriteBytes) {
+                if (messageHashCode != lastMessageHashCode ||
+                    pendingWriteBytes != lastPendingWriteBytes || flushProgress != lastFlushProgress) {
```

Comment:
I would perform the assignment to flushProgress and the check directly on the if expression to avoid the currentProgress() load to happen regardless the pendingWriteBytes value

Reply:
Do you mean to change it to this way ?
```java
if (messageHashCode != lastMessageHashCode ||
                    pendingWriteBytes != lastPendingWriteBytes ||  buf.currentProgress() != lastFlushProgress)
```
This helps to reduce calls to the `currentProgress()` method, but in the if block we need to assign the current progress to `lastFlushProgress `, so there will be an additional call.

How about changing it to this way? Although poor readability.

```java
long flushProgress = lastFlushProgress;
if (messageHashCode != lastMessageHashCode ||
                    pendingWriteBytes != lastPendingWriteBytes ||  (flushProgress = buf.currentProgress()) != lastFlushPorgress) {
    lastFlushPorgress = flushProgress;
}
```

Reply:
Sorry for my bad explanation of the change request :P
Yes I was meaning exactly the second case you have shown, lgtm

Reply:
as the is not a volatile access I think I would just keep it as it is as it is easier to maintain .

Reply:
Or alternative (which may be better) just use a second if block in case the previous branch is true.

Reply:
Okay, let me update it.

Reply:
thanks a lot! Just make sure you also make it pass in terms of check style as your code has some check style problems ;)